rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -376,6 +378,27 @@ pub fn cast(array: &ArrayRef, to_type: &DataType) -> Result<ArrayRef> {
             Int64 => cast_string_to_numeric::<Int64Type>(array),
             Float32 => cast_string_to_numeric::<Float32Type>(array),
             Float64 => cast_string_to_numeric::<Float64Type>(array),
+            Date32(DateUnit::Day) => {
+                use chrono::{NaiveDate, NaiveTime};
+                let zero_time = NaiveTime::from_hms(0, 0, 0);
+                let string_array = array.as_any().downcast_ref::<StringArray>().unwrap();
```

Comment:
It would be nice to use `expect` rather than `unwrap` here so that we can provide an error message e.g. `.expect("failed to downcast to string in CAST from String to Date")`

Reply:
@andygrove thank you for taking the time to review this change; I absolutely agree with limiting use of unwrap() in general, but in this case because of `match (from_type, to_type) {` and then the specific case of `(Utf8, _) => match to_type {` it is unlikely that the cast of the array to cast to a `StringArray` will fail; or am I missing something?

Reply:
I took a look at the code leading up to this and can see that the data type is coming from the array, so yes, I agree that this should be unlikely to fail :+1: